tools/psa/build_tfm.py

```diff
@@ -160,22 +181,301 @@ def clean_up_cloned_repos():
+    return tuple([TARGET_MAP[target].name,
+                  TARGET_MAP[target].tfm_target_name,
+                  tc,
+                  delivery_dir,
+                  bl_sup])
+
+def get_mbed_supported_tfm_targets():
+    tfm_secure_targets = get_tfm_secure_targets()
+    logger.debug("Found the following TF-M targets: {}".format(
+                                                ', '.join(tfm_secure_targets)))
+
+    return [get_target_info(t) for t in tfm_secure_targets]
+
+def commit_changes(directory, target=None, toolchain=None):
+    """
+    Check for changes in `directory` and if any then commit them
+    :param directory: path to be checked for changes
+    :param target: Target name
+    :param toolchain: Toolchain name
+    """
+    # Use --intent-to-add option of git status to identify un-tracked files
+    cmd = ['git', '-C', ROOT, 'status', 'N', directory]
+    run_cmd_and_return_errorcode(cmd)
+
+    cmd = ['git', '-C', ROOT, 'diff', '--exit-code', '--quiet', directory]
+    changes_made = run_cmd_and_return_errorcode(cmd)
+
+    if changes_made:
+        if target:
+            logger.info("Change in image for %s has been detected" % target)
+        else:
+            logger.info("Change detected in directory %s" % directory)
+        cmd = ['git', '-C', ROOT, 'add', relpath(directory, ROOT)]
+        run_cmd_and_return_errorcode(cmd)
+
+        if target:
+            logger.info("Committing image for %s" % target)
+            msg = '--message="Updated secure binary for %s (%s)"' % (target,
+                                                                    toolchain)
+        else:
+            logger.info("Committing changes in directory %s" % directory)
+            msg = '--message="Updated directory %s "' % directory
+
+        cmd = ['git', '-C', ROOT, 'commit', msg]
+        run_cmd_and_return_errorcode(cmd)
+    else:
+        if target:
+            logger.info("No changes detected in %s, skipping commit" % target)
+        else:
+            logger.info("No changes detected in %s, skipping commit" %
+                                                    relpath(directory, ROOT))
+
+def run_cmake_build(configure, cmake_build_dir, bl_supported= None,
+                    tfm_target=None, tfm_toolchain=None, debug=False):
+
+    if configure:
+        cmake_cmd = ['cmake', '-GUnix Makefiles']
+        cmake_cmd.append('-DPROJ_CONFIG=' + (join(TF_M_BUILD_DIR,
+                         'trusted-firmware-m', 'configs/ConfigCoreIPC.cmake')))
+        cmake_cmd.append('-DTARGET_PLATFORM=' + tfm_target)
+        cmake_cmd.append('-DCOMPILER=' + tfm_toolchain)
+        if debug:
+            cmake_cmd.append('-DCMAKE_BUILD_TYPE=Debug')
+        else:
+            cmake_cmd.append('-DCMAKE_BUILD_TYPE=Release')
+        if not bl_supported:
+            cmake_cmd.append('-DBL2=False')
+        cmake_cmd.append('..')
+    else:
+        cmake_cmd = ['cmake', '--build', '.', '--', 'install']
```

Comment:
What does `install` do?

Reply:
@Patater Export NS APIs to a dedicated folder under cmake build folder

Reply:
Add a comment please.

Reply:
Fixed.